server/application-server/src/main/java/de/tum/cit/aet/helios/error/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,93 @@
+import de.tum.cit.aet.helios.environment.EnvironmentException;
+import de.tum.cit.aet.helios.github.permissions.PermissionException;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.time.Instant;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+  // -- 404 NOT FOUND -----------------------------------
+  @ExceptionHandler(EntityNotFoundException.class)
+  public ResponseEntity<ApiError> handleEntityNotFoundException(
+      EntityNotFoundException ex, HttpServletRequest request) {
+
+    ApiError error = new ApiError();
+    error.setStatus(HttpStatus.NOT_FOUND.value());
+    error.setError(HttpStatus.NOT_FOUND.getReasonPhrase());
+    error.setMessage("Error: " + ex.getMessage());
+    error.setPath(request.getRequestURI());
+    error.setTimestamp(Instant.now());
+
+    return new ResponseEntity<>(error, HttpStatus.NOT_FOUND);
+  }
+
+  // -- 400 BAD REQUEST ----------------------------------
+  // Combining IllegalStateException & IllegalArgumentException
+  @ExceptionHandler({IllegalStateException.class, IllegalArgumentException.class})
+  public ResponseEntity<ApiError> handleBadRequestExceptions(
+      RuntimeException ex, HttpServletRequest request) {
+
+    ApiError error = new ApiError();
+    error.setStatus(HttpStatus.BAD_REQUEST.value());
+    error.setError(HttpStatus.BAD_REQUEST.getReasonPhrase());
+    error.setMessage("Error: " + ex.getMessage());
+    error.setPath(request.getRequestURI());
+    error.setTimestamp(Instant.now());
+
+    return new ResponseEntity<>(error, HttpStatus.BAD_REQUEST);
+  }
+
+  // -- 403 FORBIDDEN ------------------------------------
+  @ExceptionHandler({SecurityException.class, PermissionException.class, 
+      EnvironmentException.class})
+  public ResponseEntity<ApiError> handleSecurityExceptions(
+      RuntimeException ex, HttpServletRequest request) {
+
+    ApiError error = new ApiError();
+    error.setStatus(HttpStatus.FORBIDDEN.value());
+    error.setError(HttpStatus.FORBIDDEN.getReasonPhrase());
+    error.setMessage("Error: " + ex.getMessage());
+    error.setPath(request.getRequestURI());
+    error.setTimestamp(Instant.now());
+
+    return new ResponseEntity<>(error, HttpStatus.FORBIDDEN);
+  }
+
+  // -- 500 INTERNAL SERVER ERROR (FALLBACK) -------------
+  @ExceptionHandler({Exception.class, IOException.class})
+  public ResponseEntity<ApiError> handleGeneralException(
+      Exception ex, HttpServletRequest request) {
+
+    ApiError error = new ApiError();
+    error.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
+    error.setError(HttpStatus.INTERNAL_SERVER_ERROR.getReasonPhrase());
+    error.setMessage("Error: " + ex.getMessage());
```

Comment:
issue: not sure about sending the exception message to the client and show it there as it might leak sensitive information